### DISCUSSION TITLE
Accessible brand color for light and dark themes

### DISCUSSION THREAD

**Discussion by ludwiczakpawel** [forem/forem#16161] [Feature Requests]
Currently admins can specify only one brand color.

![image](https://user-images.githubusercontent.com/108287/149909831-bb9137db-aafb-4387-a894-076c8b887ea5.png)

This color is primarily being used to style buttons, links and few other accents. When admin specifies the color we check the contrast ratio against white color and throw an error if it's insufficient. 

This only works in light theme. For dark theme we ignore the brand color and use some hardcoded one. But we do want to let dark theme to use brand color as well. But because of our contrast verification the picked brand color may not work on dark theme (just pick any dark color and it simply won't be visible against dark background) - example: see this primary "Publish" button which should be vivid but it's not:

![image](https://user-images.githubusercontent.com/108287/149911034-a6b2e86e-42e7-4417-bedb-98d633337da8.png)

---

One of the potential solutions is to let admins specify two colors: one brand color for light theme and one for dark theme. And this could potentially be our starting point. And it's worth pointing out this already kinda happens: we will soon let admins upload separate logo for light and dark themes so why not letting them specify separate colors for both themes as well?

There's a caveat though: we eventually will let admins specify a bit more colors, e.g. header background color, separate color for buttons and links, etc... So we need some sort of customization and flexibility here. But the more we let them customize, the more complex it will become because each specified color will have to be defined twice - for light and dark theme. 

---

_So dear diary, what should we do..?_

**Comment by Ridhwana** [comment]
> One of the potential solutions is to let admins specify two colors: one brand color for light theme and one for dark theme. And this could potentially be our starting point. And it's worth pointing out this already kinda happens: we will soon let admins upload separate logo for light and dark themes so why not letting them specify separate colors for both themes as well?

I like this as a starting point especially since we're already in the process of mapping out a feature that allows for separate logos to be uploaded per theme. I do think we need to improve on the information architecture a bit and eventually let these (the logos and brand colors) live closer together so that it helps creators to connect the dots better. 

My suggestion is that based on the light themed color they choose (which will most likely be during onboarding, I suspect that we won't want to clutter their onboarding with choosing dark themed values), we suggest a dark theme brand color for them and let them override it if they'd like.

> There's a caveat though: we eventually will let admins specify a bit more colors, e.g. header background color, separate color for buttons and links, etc...

I think this is also doable with the above solution. I feel like the user experience will be key here. Some creators may want to customize all of the above whilst others may not. Here's some suggestions: 

1. Provide dark theme defaults for them based on their light theme color and auto populate this for them (as per above). 
2. If they decide to enter a "dark themed brand color", we could again populate these other custom fields, like the header background etc. with that value.  
3. In some cases creators may just stick with that and in others they can then go in and fine tune these even further. 

Sincerely, 
Your Diary

**Reply by ludwiczakpawel** [reply]
100% agree with all you wrote above.

**Reply by amywtlin** [reply]
💯

**Comment by Ridhwana** [comment]
I'm also curious to know (maybe through a user study) on how many admins will want to specify even more fine tuning theming like a header background color, separate color for buttons and links, etc

I've always had the impression (and it may not be an accurate one 🤷🏽‍♀️) that a good theme is one that is consistent and maybe has less variation of colors? 🤷🏽‍♀️

I guess a balance here would be important which I'm sure product design has a good handle on anyways .😁 ... 

The End of my 2c

**Reply by ludwiczakpawel** [reply]
> I'm also curious to know (maybe through a user study) on how many admins will want to specify even more fine tuning theming like a header background color, separate color for buttons and links, etc

I don't have an exact number here but it's been a request I've heard at least several times. How granular the customization will be is up to us obviously - what we think makes the most sense. I definitely don't want to give TOO MUCH power to admin but on the other hand they really want to be able to customize _some_ things. Definitely header background and also links color being differentiated from "brand" color - since brand color is often dark enough to not be easily distinguishable from regular text (see for example forem.dev - how hard it is to spot links in posts - that dark green is just 🤷 ).

And totally agree - it's a matter of good balance.

> product design has a good handle on

lolz i dunno watchya talking bout _dog_in_front_of_computer.gif_

**Comment by ellativity** [comment]
I'm crashing this party with a suggestion that "Brand color" is a misleading name, and often leaves Creators disappointed that they can't *actually* use their **Brand color**. Have we considered renaming the current "Brand color" to "Site Emphasis Color" or something (not that though, because it's a sucky suggestion), and giving people genuine branding options?


> a good theme is one that is consistent and maybe has less variation of colors

I do agree with this as a general(ized) design principle. Most brands do have a range of colours they'd like to feature, and for them the notion of consistency is less about consistency from a generalized design perspective than consistency from the perspective of their brand.

**Reply by ludwiczakpawel** [reply]
Yea, this definitely is a bad name! :D I'm keen to rename it to anything that doesn't mention brand. "Accent color" - feels like the most appropriate language to me here but I'm not an expert 🤷‍♂️ "Site Emphasis" feels too... scientific? :D.. Nevertheless, once we enable customization of MORE colors - this is where the brand will get reflected and where (how) it's gonna be defined in general...

**Reply by amywtlin** [reply]
Another common set of terms describing this is "Primary color/Secondary color" for your reference. (Though we don't really have a secondary color)

**Reply by amywtlin** [reply]
"Accent color" also seems quite straightforward to me - I think it'll be great to consider adding some visual examples or textual description telling admins how & where this color they just set up would be used.

**Reply by ellativity** [reply]
>(Though we don't really have a secondary color)

Until we do! With MORE colors, eh, @ludwiczakpawel ?

**Reply by maestromac** [reply]
`Accent color` is also the same term used on Android devices so I believe many people will get it. Another icing on the cake would be to show preview image so the admin have a better idea what it will look like 

<img src="https://user-images.githubusercontent.com/15793250/150172435-e526a60c-4145-49e0-8dae-a64159867fa7.png" width="200" />

**Comment by saltpeaky99** [comment]
Update your WordPress to the latest version 9.0 which was released recently. My client is also facing this error while posting in their [Wordle Junior ](https://wordle-unlimited.us/wordle-junior/). The New version is awesome must try it.
